common/models/role-mapping.js

```diff
@@ -24,9 +24,11 @@ module.exports = function(RoleMapping) {
    * @param {Application} application
    */
   RoleMapping.prototype.application = function(callback) {
+    var registry = this.constructor.registry || loopback.registry;
```

Comment:
Most of the other code relies on the fact that `ModelCtor.registry` is always set. What is the reason for using `|| loopback.registry` in this method? If `this.constructor.registry` is always set, then the second branch is redundant IMO.